### PR TITLE
add test for #353

### PR DESCRIPTION
**Pull request by SimonLab**
see [regression test comment](https://github.com/FAC-GM/app/pull/354#issuecomment-200297473)

### CODE REVIEW

File: test/z-clients.test.js [hunk 1]
```diff
@@ -168,6 +169,52 @@ describe('Attempt to save/update a client: /clients/0 without authorization', fu
...
+      server.inject(options , function (res) {
+        expect(err).to.not.exist();
+        expect(res.statusCode).to.equal(302);
+
+        //wait for ES to index the new value
+        setInterval(function() {
+        //check that the possibleNames is still an empty array
+          getClient(0,function(err, client) {
+
+            expect(client.possibleNames.length).to.equal(0);
```
Comment (nelsonic) [inline test/z-clients.test.js]: so possible names is a string of values?
and this is testing that the system does not fail if its *unset*?

File: test/z-clients.test.js [hunk 2]
```diff
@@ -168,6 +169,52 @@ describe('Attempt to save/update a client: /clients/0 without authorization', fu
...
+    };
+
+    Server.init(0, function (err, server) {
+
+      server.inject(options , function (res) {
+        expect(err).to.not.exist();
+        expect(res.statusCode).to.equal(302);
+
+        //wait for ES to index the new value
+        setInterval(function() {
```
Comment (nelsonic) [inline test/z-clients.test.js]: Why setInterval?
How many times do you need to call this?

Comment (SimonLab) [inline test/z-clients.test.js]: Good catch, I wanted to use setTimeout()

File: test/z-clients.test.js (hunk 1 above)
Comment (SimonLab) [inline test/z-clients.test.js]: possible names is an array of string (the values from the form are separate with by a comma). The error was if the field is empty an empty string is added to the possible names value

Comment (nelsonic) [inline test/z-clients.test.js]: If `request.payload.possibleNames` is *unset* the server will ***throw***: 
```sh
TypeError: Cannot read property 'split' of undefined
```
see: https://repl.it/ByvF/1
so...
https://github.com/FAC-GM/app/blob/1e1af2de415995493066659304febe77804cd419/lib/handlers/clients/post_save.js#L14
```js
var possibleNames = request.payload.possibleNames.split(',');
```
should be:
```js
var possibleNames = request.payload.possibleNames ? request.payload.possibleNames.split(',') : [];
```
see: https://repl.it/ByvF/2 *no type error*.

Comment (nelsonic) [inline test/z-clients.test.js]: Or...
```js
var possibleNames = !request.payload.possibleNames ? [] : request.payload.possibleNames.split(',')
    .map(name => {return name.trim()}).filter(Boolean);
```
https://repl.it/ByvF/5

**Comment by nelsonic** [comment]
@SimonLab I don't know if you saw my other comment... 
https://github.com/FAC-GM/app/pull/356#discussion_r57166746

File: test/z-clients.test.js (hunk 1 above)
Comment (SimonLab) [inline test/z-clients.test.js]: possibleNames will always be define from the form (https://github.com/FAC-GM/app/blob/4a6d8568f0229310c134c76ee124cc75ba661875/views/clientFormView.html#L13). The only problem was that the value could be an empty string which we don't want to add to the array possibleNames. The only way the value possibleNames could be undefined is if someone try to save a client without passing by the form (using curl for example with an empty payload). I'm not sure if I'm clear or if I understood properly your previous comment

Comment (nelsonic) [inline test/z-clients.test.js]: Ok. given that this route is "*protected*" by auth and the client form always sends a value, we don't *have* to worry about its value... :+1:

**Comment by SimonLab** [comment]
:+1:

**Pull request merged**